benchmark/worker/benchmark_client.go

```diff
@@ -0,0 +1,331 @@
+		if err != nil {
+			return nil, grpc.Errorf(codes.InvalidArgument, "failed to create TLS credentials %v", err)
+		}
+		opts = append(opts, grpc.WithTransportCredentials(creds))
+	} else {
+		opts = append(opts, grpc.WithInsecure())
+	}
+
+	grpclog.Printf(" - core limit: %v", config.CoreLimit)
+	// Use one cpu core by default.
+	// TODO: change default number of cores used if 1 is not fastest.
+	if config.CoreLimit > 1 {
+		runtime.GOMAXPROCS(int(config.CoreLimit))
+	}
+
+	grpclog.Printf(" - payload config: %v", config.PayloadConfig)
+	var (
+		payloadReqSize, payloadRespSize int
+		payloadType                     string
+	)
+	if config.PayloadConfig != nil {
+		switch c := config.PayloadConfig.Payload.(type) {
+		case *testpb.PayloadConfig_BytebufParams:
+			opts = append(opts, grpc.WithCodec(byteBufCodec{}))
+			payloadReqSize = int(c.BytebufParams.ReqSize)
+			payloadRespSize = int(c.BytebufParams.RespSize)
+			payloadType = "bytebuf"
+		case *testpb.PayloadConfig_SimpleParams:
+			payloadReqSize = int(c.SimpleParams.ReqSize)
+			payloadRespSize = int(c.SimpleParams.RespSize)
+			payloadType = "protobuf"
+		case *testpb.PayloadConfig_ComplexParams:
+			return nil, grpc.Errorf(codes.Unimplemented, "unsupported payload config: %v", config.PayloadConfig)
+		default:
+			return nil, grpc.Errorf(codes.InvalidArgument, "unknow payload config: %v", config.PayloadConfig)
+		}
+	}
+
+	grpclog.Printf(" - rpcs per chann: %v", config.OutstandingRpcsPerChannel)
+	grpclog.Printf(" - channel number: %v", config.ClientChannels)
+
+	rpcCountPerConn, connCount := int(config.OutstandingRpcsPerChannel), int(config.ClientChannels)
+
+	grpclog.Printf(" - load params: %v", config.LoadParams)
+	// TODO add open loop distribution.
+	switch config.LoadParams.Load.(type) {
+	case *testpb.LoadParams_ClosedLoop:
+	case *testpb.LoadParams_Poisson:
+		return nil, grpc.Errorf(codes.Unimplemented, "unsupported load params: %v", config.LoadParams)
+	case *testpb.LoadParams_Uniform:
+		return nil, grpc.Errorf(codes.Unimplemented, "unsupported load params: %v", config.LoadParams)
+	case *testpb.LoadParams_Determ:
+		return nil, grpc.Errorf(codes.Unimplemented, "unsupported load params: %v", config.LoadParams)
+	case *testpb.LoadParams_Pareto:
+		return nil, grpc.Errorf(codes.Unimplemented, "unsupported load params: %v", config.LoadParams)
+	default:
+		return nil, grpc.Errorf(codes.InvalidArgument, "unknown load params: %v", config.LoadParams)
+	}
+
+	grpclog.Printf(" - rpc type: %v", config.RpcType)
+	var rpcType string
+	switch config.RpcType {
+	case testpb.RpcType_UNARY:
+		rpcType = "unary"
+	case testpb.RpcType_STREAMING:
+		rpcType = "streaming"
+	default:
+		return nil, grpc.Errorf(codes.InvalidArgument, "unknown rpc type: %v", config.RpcType)
+	}
+
+	grpclog.Printf(" - histogram params: %v", config.HistogramParams)
+	grpclog.Printf(" - server targets: %v", config.ServerTargets)
+
+	conns := make([]*grpc.ClientConn, connCount)
```

Comment:
you can move line 157 - 173 above the switch on line 145 so that you do not need to switch twice.